hips/hip-00XX.md

```diff
@@ -0,0 +1,136 @@
+
+## Specification
+
+<!-- Describe the syntax and semantics of any new feature. -->
+
+### General
+
+Helm will add a new flag to the install, upgrade and rollback sub-commands, modelled after the like `kubectl` flag. And add a corresponding field to the respective install/upgrade/rollback SDK API objects:
+`--server-side=false|true|auto`
+
+Helm will default to `false` for installs, retaining its existing behaviour. When `true`, Helm will submit resources to the Kubernetes API server with SSA enabled. When `auto`, Helm will default to CSA on installs (`false`), and for upgrades and rollbacks enable SSA based on the whether SSA was utilized for the prior operation (`auto`). The usage of SSA for a given release will be stored in the release's manifest.
+
+If a user deploys a chart with a new version of Helm with SSA enabled, then upgrades the chart with an older version of Helm. It is expected that the older Helm will be able to upgrade the chart with CSA. And the new manifest object will omit the indication of SSA usage from the releases manifest (and so futher upgrades/rollbacks will continue to use CSA by default).
+
+In a future version of Helm 3 cli, it might be desirable to omit a warning if a user is not using SSA (in order to prompt the user to enabling SSA).
+
+#### Custom resources
+
+Helm will utilize SSA for custom resources, which previously it replaced (TODO: reference). As SSA allows custom resources to be correctly merged based on metadata within the CRD. Or the API server will use the default approach <https://kubernetes.io/docs/reference/using-api/server-side-apply/#custom-resources>.
+
+#### Conflicts and forcing
+
+It is possible when Helm upgrades a chart, that there will be a field conflict with another field manager. In this case, Helm should report the error to the user.
+
+Helm will also add a new flag `--force-conflicts`, and corresponding field to the `upgrade` and `rollback` commands. Which will tell the apply to ignore/override field conflicts (and so thus make Helm the field manager).
+
+(Of note, Helm upgrade already includes the `--force` flag, which tells Helm to replace objects <TODO: confirm this, going from memory>. However, given this different semantics, it is thought a new flag is required)
+
+### Implementation
+
+<TODO: how do we execute SSA>
+- field manager name
+
+https://github.com/kubernetes/kubectl/blob/197123726db24c61aa0f78d1f0ba6e91a2ec2f35/pkg/cmd/apply/apply.go#L248
+https://github.com/kubernetes/kubectl/blob/197123726db24c61aa0f78d1f0ba6e91a2ec2f35/pkg/cmd/apply/apply.go#L564-L657
+
+#### Special
+
+- `--dry-run=client` doesn't work with SSA enabled
```

Comment:
@mattfarina points out that this will make it difficult for user's to preview changes